client/src/components/Filters/PriceFilter.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import ToggleButton from "@mui/material/ToggleButton";
+import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
+
+export default function PriceFilterToggler() {
+  const DEFAULT_FILTER = {
+    $: false,
+    $$: false,
+    $$$: false,
+    $$$$: false,
+  };
+
+  const [priceFilter, setPriceFilter] = React.useState(DEFAULT_FILTER);
+  const prices = ["$", "$$", "$$$", "$$$$"];
+
+  return (
+    <ToggleButtonGroup aria-label="price" size="small">
+      {prices.map((price) => (
+        <ToggleButton
+          onClick={() => {
+            const updatedFilter = {
+              ...priceFilter,
+              $: !priceFilter["$"],
```

Comment:
That almost works. The reason it failed is because you're assigning literally "price" to be the key. Whereas you intend to assign the value of price to be the key.

Try this:
```js
[price]: !priceFilter[price]
```